File: rb/features.rb
# Techy SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module TechyFeatures
  def self.make_feature(name)
    case name
    when "base"
      TechyBaseFeature.new
    when "test"
      TechyTestFeature.new
    else
      TechyBaseFeature.new
    end
  end
end

File: py/techy_sdk/features.py
# Techy SDK feature factory

from techy_sdk.feature.base_feature import TechyBaseFeature
from techy_sdk.feature.test_feature import TechyTestFeature


def _make_feature(name):
    features = {
        "base": lambda: TechyBaseFeature(),
        "test": lambda: TechyTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
